tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -155,7 +163,56 @@ object NativeConfig {
     def withOptimize(value: Boolean): NativeConfig =
       copy(optimize = value)
 
-    override def toString: String =
+    override def withLinktimeProperties(v: Map[String, Any]): NativeConfig = {
+      def isNumberOrString(value: Any) = {
+        def isNonUnitPrimitive = value match {
+          case _: java.lang.Boolean | Boolean => true
```

Comment:
This seems extremely weird. It would match any instance of `java.lang.Boolean` (which also means any `Boolean`), or *the companion object `scala.Boolean`*. I'm pretty sure that is not what was intended.

I think what you intended was
```scala
def isNonUnitPrimitive = value match {
  case _: Byte | _: Char | _: Short | _: Int | _: Long | _: Float | _: Double => true
  case                                                                        => false
}
```